Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
Hence, this assert. The exit-peer does not need to forward it/process it as it receives it over the exit-socket. All packets received by this socket are automatically forwarded back to the originator. And hence the check_key_response method is only called at the originator and not at the exit-peer.